### PR TITLE
(SERVER-1494) introduce lein parent project

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

**Comment by cprice404** [comment]
This is blocked on puppetlabs/trapperkeeper#257

**Comment by camlow325** [comment]
👍 when we have a TK release that can be included from `clj-parent`.  I built one locally from the current HEAD of the Trapperkeeper branch, ran the tests, and saw everything pass.  I also did a pass back through the dependencies to see that the ones being replaced via `clj-parent` would all be the same or later than what tk-jetty9 is currently picking up.

File: project.clj [hunk 1]
```diff
@@ -71,27 +66,20 @@
...
-                   :dependencies [[puppetlabs/http-client "0.5.0"]
-                                  [puppetlabs/kitchensink ~ks-version :classifier "test"]
-                                  [puppetlabs/trapperkeeper ~tk-version :classifier "test"]
-                                  [org.clojure/tools.namespace "0.2.10"]
-                                  [spyscope "0.1.4"]
-                                  [compojure "1.1.8" :exclusions [ring/ring-core
-                                                                  commons-io
-                                                                  org.clojure/tools.macro]]
+                   :dependencies [[puppetlabs/http-client]
+                                  [puppetlabs/kitchensink nil :classifier "test"]
```
Comment (cprice404) [inline project.clj]: Just gonna leave 'em for now, Mr. Rock.

**Comment by cprice404** [comment]
Restarted travis now that TK is released.

**Comment by cprice404** [comment]
This PR is blocking other work so I'm probably going to self-merge if no one has time to review it.

**Comment by camlow325** [comment]
Sounds like no other takers.  I'm going to go ahead and merge this.

**Pull request merged**